cpu/sam0_common/periph/gpio.c

```diff
@@ -220,17 +239,70 @@ void gpio_irq_disable(gpio_t pin)
     if (exti == -1) {
         return;
     }
-    EIC->INTENCLR.reg = (1 << exti);
+    _EIC->INTENCLR.reg = (1 << exti);
+}
+
+#ifdef CPU_SAML1X
+void isr_eic0(void)
```

Comment:
I would do this differently do avoid all this code duplication:

```c
#ifdef CPU_SAML1X
void isr_eic0(void)
{
    isr_eic()
}
...
void isr_eic1(void)
{
    isr_eic()
}

void isr_eic_other(void)
{
    isr_eic()
}
#endif /* SAML1X */

void isr_eic(void)
{
    for (unsigned i = 0; i < NUMOF_IRQS; i++) {
        if (_EIC->INTFLAG.reg & (1 << i)) {
            _EIC->INTFLAG.reg = (1 << i);
            gpio_config[i].cb(gpio_config[i].arg);
        }
    }
    cortexm_isr_end();
}
```

This is untested but I think it should work.

Reply:
I'll give it a try.